### PR TITLE
Migrate Knowledgemotion to common-ingest

### PR DESCRIPTION
**Pull request by williamberg**

### CODE REVIEW

File: src/main/java/org/atlasapi/remotesite/knowledgemotion/KnowledgeMotionCsvTranslator.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            String source = record.get(KnowledgeMotionSpreadsheetColumn.SOURCE.getFieldName());
+            String id = record.get(KnowledgeMotionSpreadsheetColumn.ID.getFieldName());
+            String title = record.get(KnowledgeMotionSpreadsheetColumn.TITLE.getFieldName());
+            String description = record.get(KnowledgeMotionSpreadsheetColumn.DESCRIPTION.getFieldName());
+            String date = record.get(KnowledgeMotionSpreadsheetColumn.DATE.getFieldName());
+            String duration = record.get(KnowledgeMotionSpreadsheetColumn.DURATION.getFieldName());
+            String keywordsString = record.get(KnowledgeMotionSpreadsheetColumn.KEYWORDS.getFieldName());
+            String altId = record.get(KnowledgeMotionSpreadsheetColumn.ALT_ID.getFieldName());
+
+            List<String> keywords = new ArrayList<>();
```
Comment (tfm) [inline src/main/java/org/atlasapi/remotesite/knowledgemotion/KnowledgeMotionCsvTranslator.java]: Can we use an `ImmutableList.builder` here?

File: src/main/java/org/atlasapi/remotesite/knowledgemotion/KnowledgeMotionDataRow.java [hunk 2]
```diff
@@ -17,12 +17,8 @@
...
     private final List<String> keywords;
     private final String alternativeId;
-    
-    public static Builder builder() {
-        return new Builder();
-    }
-    
-    private KnowledgeMotionDataRow(String source, String id, String title, String description, 
+
+    public KnowledgeMotionDataRow(String source, String id, String title, String description,
```
Comment (tfm) [inline src/main/java/org/atlasapi/remotesite/knowledgemotion/KnowledgeMotionDataRow.java]: Not sure we should have removed the builder in favour of a many-arguments-the-same-type-and-therefore-error-prone constructor

File: src/main/java/org/atlasapi/remotesite/knowledgemotion/KnowledgeMotionFileProcessor.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+    private final ContentResolver contentResolver;
+    private final ContentWriter contentWriter;
+    private final ContentLister contentLister;
+    private final TopicGuesser topicGuesser;
+    private final KnowledgeMotionCsvTranslator csvTranslator;
+
+    public KnowledgeMotionFileProcessor(ContentResolver contentResolver, ContentWriter contentWriter,
+        ContentLister contentLister, TopicGuesser topicGuesser) {
+        this.contentResolver = contentResolver;
```
Comment (tfm) [inline src/main/java/org/atlasapi/remotesite/knowledgemotion/KnowledgeMotionFileProcessor.java]: `checkNotNull`

File: src/main/java/org/atlasapi/remotesite/knowledgemotion/KnowledgeMotionFileProcessor.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        this.contentWriter = contentWriter;
+        this.contentLister = contentLister;
+        this.topicGuesser = topicGuesser;
+
+        csvTranslator = new KnowledgeMotionCsvTranslator();
```
Comment (tfm) [inline src/main/java/org/atlasapi/remotesite/knowledgemotion/KnowledgeMotionFileProcessor.java]: We've passed in all other collaborators, why not this one?

File: src/main/java/org/atlasapi/remotesite/knowledgemotion/KnowledgeMotionModule.java [hunk 5]
```diff
@@ -15,25 +13,33 @@
...
-
     private @Autowired DatabasedMongo mongo;
 
+    @Value("${km.contentdeals.aws.accessKey}")
+    private String awsAccessKey;
+    @Value("${km.contentdeals.aws.secretKey}")
+    private String awsSecretKey;
+    @Value("${km.contentdeals.aws.s3BucketName}")
+    private String awsS3BucketName;
+    @Value("${km.contentdeals.aws.sqsQueueName}")
```
Comment (tfm) [inline src/main/java/org/atlasapi/remotesite/knowledgemotion/KnowledgeMotionModule.java]: How come we need to know the SQS queue name? Isn't that private to the `common-ingest` streamer?

File: src/main/java/org/atlasapi/remotesite/knowledgemotion/KnowledgeMotionSpreadsheetColumn.java [hunk 6]
```diff
@@ -4,7 +4,7 @@
 
 public enum KnowledgeMotionSpreadsheetColumn {
 
-    SOURCE("source"),
+    SOURCE("Source"),
     ID("namespace"),
```
Comment (tfm) [inline src/main/java/org/atlasapi/remotesite/knowledgemotion/KnowledgeMotionSpreadsheetColumn.java]: Can we have consistent capitalisation in the column headings?

File: src/main/java/org/atlasapi/remotesite/knowledgemotion/KnowledgeMotionUpdater.java [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+                if (written.isPresent()) {
+                    seenUris.add(written.get().getCanonicalUri());
+                }
+            } catch (RuntimeException e) {
+                processingResult.error(row.getId(), e.getMessage());
+            }
+        }
+
+        // un-ActivelyPublisheding disappeared content
+        Iterator<Content> allStoredKmContent = contentLister.listContent(ContentListingCriteria.defaultCriteria().forContent(ContentCategory.TOP_LEVEL_ITEM).forPublishers(allKmPublishers).build());
```
Comment (tfm) [inline src/main/java/org/atlasapi/remotesite/knowledgemotion/KnowledgeMotionUpdater.java]: Let's not call this if we have failed to write any rows

File: src/main/java/org/atlasapi/remotesite/knowledgemotion/KnowledgeMotionSpreadsheetColumn.java (hunk 6 above)
Comment (williamberg) [inline src/main/java/org/atlasapi/remotesite/knowledgemotion/KnowledgeMotionSpreadsheetColumn.java]: Unfortunately not. The header row is `Source,Unique ID,namespace,Title,Description,Date,Duration,Keywords,"Price category (1= stock, 2=news, 3=brand)",Sounds,Color,Location,Country,State,City,Region,Alternative ID,Source`

Comment (tfm) [inline src/main/java/org/atlasapi/remotesite/knowledgemotion/KnowledgeMotionSpreadsheetColumn.java]: Can we make it case insensitive, then?

**Pull request closed without merging**